packages/wrangler/src/r2/helpers.ts

```diff
@@ -88,6 +115,28 @@ export function bucketAndKeyFromObjectPath(objectPath = ""): {
 	return { bucket: match[1], key: match[2] };
 }
 
+/**
+ * Fetch a list of all the objects under the given `accountId` and `bucketName`.
+ */
+export async function listR2Objects(
+	accountId: string,
+	bucketName: string,
+	params: ListObjectsRequest
+): Promise<ListObjectsResponse> {
+	const response = await fetchR2Objects(
+		`/accounts/${accountId}/r2/buckets/${bucketName}/objects?per_page=${
+			params.per_page ?? 1000
+		}${params.cursor ? "&cursor=" + encodeURIComponent(params.cursor) : ""}${
+			params.prefix ? "&prefix=" + encodeURIComponent(params.prefix) : ""
+		}${
+			params.start_after
+				? "&start_after=" + encodeURIComponent(params.start_after)
+				: ""
+		}`
+	);
```

Comment:
Tiny nit, but you can use `new URLSearchParams()` here to make things explicit (it won't help with the encoding we need to do here so it may not be worth it, up to you!)

Reply:
kept it as it is since not much benefit